beacon-chain/rpc/service.go

```diff
@@ -35,14 +38,15 @@ type Config struct {
 
 // NewRPCService creates a new instance of a struct implementing the BeaconServiceServer
 // interface.
-func NewRPCService(ctx context.Context, cfg *Config) *Service {
+func NewRPCService(ctx context.Context, cfg *Config, announcer types.CanonicalEventAnnouncer) *Service {
```

Comment:
Can the announcer be part of the config?

Reply:
I run into weird conditions when testing because I send to a channel in the announcer interface implementation if I put it as part of config, as it seems it creates a copy that is different in the RPCService struct if I do that

Reply:
We can explore this in a separate PR if this service grows beyond 3 arguments